File: LeetCode/875. Koko Eating Bananas/solution.swift
import Foundation

class Solution {
    func minEatingSpeed(_ piles: [Int], _ h: Int) -> Int {
        var left = 1
        var right = piles.max() ?? 1

        while left < right {
            let middle = (left + right) / 2
            var hourSpent = 0

            for pile in piles {
                hourSpent += Int(ceil(Double(pile) / Double(middle)))
            }

            if hourSpent <= h {
                right = middle
            } else {
                left = middle + 1
            }
        }

        return right
    }
}

File: LeetCode/875. Koko Eating Bananas/solution.py
from typing import List
import math


class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        left = 1
        right = max(piles)

        while right > left:
            count = 0
            mid = (right + left) // 2

            for i in piles:
                count += math.ceil(i / mid)

            if h >= count:
                right = mid
            else:
                left = mid + 1

        return right
